### PR TITLE
Sphere mesh generator

### LINKED ISSUE #22591: Sphere mesh generator
## Reason
Many applications could benefit from a sphere volume mesh generator in MOOSE. libMesh has a mesh generator, but MOOSE does not have an interface to it.

## Design
Create `SphereMeshGenerator` that interfaces to libMesh.

## Impact
New feature.

### PR DESCRIPTION
**Pull request by aprilnovak**
Adds sphere mesh generator.

Closes #22591

### CODE REVIEW

**Comment by GiudGiud** [comment]
@friedmud I think you worked on a sphere mesh at some point. Any comment on that one?
anyone else who would have an opinion or needs to know?

File: test/tests/meshgenerators/sphere_mesh_generator/smooth.i [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+    type = SphereMeshGenerator
+    radius = 1.0
+    nr = 2
+    n_smooth = 1
+  []
+[]
+
+[Outputs]
+  exodus = true
+[]
```
Comment (GiudGiud) [inline test/tests/meshgenerators/sphere_mesh_generator/smooth.i:12]:
```suggestion
```

File: test/tests/meshgenerators/sphere_mesh_generator/sphere.i [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  [sphere]
+    type = SphereMeshGenerator
+    radius = 1.0
+    nr = 2
+  []
+[]
+
+[Outputs]
+  exodus = true
+[]
```
Comment (GiudGiud) [inline test/tests/meshgenerators/sphere_mesh_generator/sphere.i:11]:
```suggestion
```

File: test/tests/meshgenerators/sphere_mesh_generator/sphere_hex27.i [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+    type = SphereMeshGenerator
+    radius = 1.0
+    nr = 2
+    elem_type = HEX27
+  []
+[]
+
+[Outputs]
+  exodus = true
+[]
```
Comment (GiudGiud) [inline test/tests/meshgenerators/sphere_mesh_generator/sphere_hex27.i:12]:
```suggestion
```

**Review by GiudGiud — CHANGES_REQUESTED** [review]
the image needs to live in large_media 

the exodus output in the test files arent needed

**Comment by aprilnovak** [comment]
Does large_media use git lfs? Or do I just commit the png image there?

**Comment by GiudGiud** [comment]
no it s just regular git.
you just add a commit on top of master and push

**Review by GiudGiud — DISMISSED** [review]

**Review by GiudGiud — APPROVED** [review]

**Comment by GiudGiud** [comment]
it is
thanks for the PR!
are you going to be meshing around the spheres in MOOSE or is this for fuel sims?

**Comment by aprilnovak** [comment]
This is for heat conduction cases in pebble beds. We have some mesh scripts at ANL to do the fluid space, but it's  much more challenging. We could potentially convert those to MOOSE functionalities, but I'd need to check with the authors.

**Pull request merged**